client/components/domains/register-domain-step/index.jsx

```diff
@@ -289,27 +303,29 @@ var RegisterDomainStep = React.createClass( {
 						}
 
 						this.props.onDomainsAvailabilityChange( true );
-
 						callback( null, result );
 					} );
 				},
 				callback => {
 					const params = {
-						quantity: SUGGESTION_QUANTITY,
-						includeWordPressDotCom: this.props.includeWordPressDotCom,
-						vendor: abtest( 'domainSuggestionVendor' )
-					};
-
+							quantity: SUGGESTION_QUANTITY,
+							includeWordPressDotCom: this.props.includeWordPressDotCom,
+							vendor: abtest( 'domainSuggestionVendor' )
+						},
+						timestamp = Date.now();
 					wpcom.fetchDomainSuggestions( domain, params, ( error, domainSuggestions ) => {
+						const timeDiff = Date.now() - timestamp;
 						if ( error && error.statusCode === 503 ) {
+							this.recordEvent( 'searchResultsReceive', domain, ['ERROR503'], timeDiff, -1, this.props.analyticsSection );
 							return this.props.onDomainsAvailabilityChange( false );
 						} else if ( error && error.error ) {
+							this.recordEvent( 'searchResultsReceive', domain, [error.code || error.error], timeDiff, -1, this.props.analyticsSection );
 							error.code = error.error;
 							this.showValidationErrorMessage( domain, error );
 						}
 
 						this.props.onDomainsAvailabilityChange( true );
-
+						this.recordEvent( 'searchResultsReceive', domain, domainSuggestions, timeDiff, domainSuggestions.length, this.props.analyticsSection );
```

Comment:
In case of an error (`error && ( error.statusCode || error.error )` checked in the `if` just above), one the search result callback will trigger two `recordEvent`s. One for the error and the second one for the returned domain suggestions (which, at best, will be empty?). Unless this is by design (I'm still wrapping my head around Tracks and all that), this line should probably go into an `else` clause for the `if` block above.